api/nakadi-event-bus-api.yaml

```diff
@@ -0,0 +1,1207 @@
+        some load balancers.
+
+        The tracking of the current position in the partitions and of which partitions is being read is in
+        the responsibility of the client.
+        No commits are needed.
+      parameters:
+        - name: name
+          in: path
+          description: Topic where to get events from
+          type: string
+          required: true
+        - name: X-nakadi-cursors
+          in: header
+          description: Array of cursors indicating the partitions to read from and the starting point on
+            each. Assumes the offset on each cursor is not inclusive (i.e., first delivered Event is the
+            **first one after** the one pointed to in the cursor).
+          required: false
+          type: array
+          items:
+            type: string
+            format: '#/definitions/Cursor'
+        - name: batch_limit
+          in: query
+          description: |
+            Maximum number of `Event`s in each chunk (and therefore per partition) of the stream.
+
+            * If 0 or unspecified will buffer Events indefinitely and flush on reaching of
+            `batch_flush_timeout`.
+          type: integer
+          format: int32
+          required: false
+          default: 1
+        - name: stream_limit
+          in: query
+          description: |
+            Maximum number of `Event`s in this stream (over all partitions being streamed in this
+            connection).
+
+            * If 0 or undefined, will stream batches indefinately.
+
+            * Stream initialization will fail if `stream_limit` is lower than `batch_limit`.
+          type: integer
+          format: int32
+          required: false
+          default: 0
+        - name: batch_flush_timeout
+          in: query
+          description: |
+            Maximum time in seconds to wait for the flushing of each chunk (per partition).
+
+            * If the amount of buffered Events reaches `batch_limit` before this batch_flush_timeout is
+            reached, the messages are immediately flushed to the client.
+
+            * If 0 or undefined, will assume 30 seconds.
+          type: integer
+          format: int32
+          required: false
+          default: 30
+        - name: stream_timeout
+          in: query
+          description: |
+            Maximum time in seconds a stream will live before being interrupted.
+            If unspecified will stream indefinately.
+
+            If this timeout is reached, any pending messages (in the sense of `stream_limit`) will be flushed
+            to the client.
+
+            Stream initialization will fail if `stream_timeout` is lower than
+            `batch_flush_timeout`.
+          type: integer
+          format: int32
+          required: false
+        - name: batch_keep_alive_limit
+          in: query
+          description: |
+            Maximum number of keep-alive messages to get in a row before closing the connection. Unlimited by
```

Comment:
@igoryankov this is a good point. 

We could introduce some optimisations in case we really see that it is a bottleneck. Usually there should not be too many partitions, we are not talking about hundreds of them.

This topic will be more of a concern when we start discussing the Subscription API. But there I hope that an application will need to keep not more then 1 connection to get all the events for the subscription (actually 1 connection for receiving and 1 connection for committing in HTTP1.1)